compiler/cgen.nim

```diff
@@ -1291,12 +1291,14 @@ proc getCopyright(conf: ConfigRef; cfile: Cfile): Rope =
         "/*   (c) " & copyrightYear & " Andreas Rumpf */$N" &
         "/* The generated code is subject to the original license. */$N" &
         "/* Compiled for: $2, $3, $4 */$N" &
+        "/* Source file: $6 */$N" &
         "/* Command for C compiler:$n   $5 */$N") %
         [rope(VersionAsString),
         rope(platform.OS[conf.target.targetOS].name),
         rope(platform.CPU[conf.target.targetCPU].name),
         rope(extccomp.CC[conf.cCompiler].name),
-        rope(getCompileCFileCmd(conf, cfile))]
+        rope(getCompileCFileCmd(conf, cfile)),
+        rope(cfile.nimname)] # ideally, put absolute path here instead
```

Comment:
That would make the nimcache files that end up in the tarball use absolute paths of the build machine. Not a great idea.

Reply:
I changed comment to
> ideally, put absolute path here instead, possibly depending on `optListFullPaths`

Reply:
Do it in this PR then please, merging this isn't urgent.

Reply:
I agree it's not urgent but I'd rather not conflate this change (which is more involved) with this PR

Reply:
Its not a conflation since you added it in this very PR.

Reply:
If you want to separate them then you can rollback the "Source file" change and focus only on the copyrights.

Reply:
done

Reply:
EDIT: see https://github.com/nim-lang/Nim/pull/15665